### PR TITLE
Serialize Get Mappings Response on Generic ThreadPool

### PR DESCRIPTION
**Pull request by original-brownbear**
Solves/mitigates the issue observed in https://github.com/elastic/elasticsearch/pull/57284

For large responses to the get mappings request, the serialization
to XContent can be extremely slow (serializing mappings is expensive since
we have to decompress and deserialize the mapping source).
To not introduce instability on the IO thread handling the get mappings response
we should move the serialization to the generic pool.
The trade-off of introducing one or two new context switches for responses that are
small enough to not cause trouble on the transport thread to prevent instability
in case of a large number of mappings in the cluster seems worth it.

Marking this team discuss for now, since we had the larger discussion of where REST actions should execute before and didn't come to a conclusion here yet. While this is a special case, we have a number of other potentially slow REST actions like this one where a very large response gets serialized.

### CODE REVIEW

**Comment by original-brownbear** [comment]
We discussed this during team discuss and decided to check if slowness in this API call happens on Cloud at any non-trivial rate.

I investigated this on Cloud and we have a lot of slow get mappings responses (up to minutes in response time) so we should take action here I think.

File: server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java [hunk 1]
```diff
@@ -62,13 +71,21 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
...
-                builder.startObject();
-                response.toXContent(builder, request);
-                builder.endObject();
-                return new BytesRestResponse(RestStatus.OK, builder);
+            protected void processResponse(GetMappingsResponse getMappingsResponse) {
+                // Process serialization on GENERIC pool since the serialization of the raw mappings to XContent can be too slow to execute
+                // on an IO thread
+                threadPool.generic().execute(ActionRunnable.wrap(this, l -> new RestBuilderListener<GetMappingsResponse>(channel) {
+                    @Override
+                    public RestResponse buildResponse(final GetMappingsResponse response, final XContentBuilder builder) throws Exception {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java:82]: WDYT about checking for a timeout again here before we do all the serialisation work? If we use a bounded threadpool then these things might pile up, so aborting early might be a helpful way to push back.

**Review by DaveCTurner — COMMENTED** [review]
I'd rather not use the generic pool for this, but `MANAGEMENT` seems like a reasonable option. See inline comment about a timeout.

File: server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java]: :+1:  how about https://github.com/elastic/elasticsearch/pull/57937/commits/0bd68d4d6be52152133889a5504bf1ff96874906 ? :)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/packaging-sample-windows

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**